src/utils.ts

```diff
@@ -33,3 +34,18 @@ export function concatNodes(nodes: EncoderNode[]): EncoderNode {
 export function escapeText(text: string) {
   return text.replace(/[.*+?^${}()|[\]\\]/g, '\\$&'); // $& means the whole matched string
 }
+
+export function extractPropsParam<Props>(
```

Comment:
Small comment of what this is used for and why we need it would be great

Reply:
I've removed that function altogether.